builder/transaction_builder.go

```diff
@@ -128,6 +272,97 @@ func (b *TransactionBuilder) BuildAndSwapToBlockBuilder(signer iotago.AddressSig
 	return blockBuilder.Payload(tx)
 }
 
+func CalculateAvailableMana(protoParams iotago.ProtocolParameters, inputSet iotago.OutputSet, targetSlot iotago.SlotIndex) (iotago.Mana, iotago.Mana, map[iotago.AccountID]iotago.Mana, error) {
+	var totalMana iotago.Mana
+	var unboundMana iotago.Mana
+	accountBoundMana := make(map[iotago.AccountID]iotago.Mana)
+
+	for inputID, input := range inputSet {
+		// calculate the potential mana of the input
+		var potentialMana iotago.Mana
+
+		// we need to ignore the storage deposit, because it doesn't generate mana
+		minDeposit := iotago.NewRentStructure(protoParams.RentParameters()).MinDeposit(input)
+		if input.BaseTokenAmount() > minDeposit {
+			excessBaseTokens, err := safemath.SafeSub(input.BaseTokenAmount(), minDeposit)
+			if err != nil {
+				return 0, 0, nil, ierrors.Wrap(err, "failed to calculate excessBaseTokens of the input")
+			}
+
+			potentialMana, err = protoParams.ManaDecayProvider().ManaGenerationWithDecay(excessBaseTokens, inputID.CreationSlot(), targetSlot)
+			if err != nil {
+				return 0, 0, nil, ierrors.Wrap(err, "failed to calculate potential mana generation and decay")
+			}
+		}
```

Comment:
```suggestion
		if excessBaseTokens, _ := safemath.SafeSub(input.BaseTokenAmount(), minDeposit); excessBaseTokens > 0 {
			potentialMana, err = protoParams.ManaDecayProvider().ManaGenerationWithDecay(excessBaseTokens, inputID.CreationSlot(), targetSlot)
			if err != nil {
				return 0, 0, nil, ierrors.Wrap(err, "failed to calculate potential mana generation and decay")
			}
		}
```

Reply:
I thinks its more clear for the reader if we do that check.